security/pkg/credentialfetcher/plugin/gce_test.go

```diff
@@ -0,0 +1,319 @@
+		jwtPath       string
+		expectedToken string
+		expectedCall  int
+		expectedErr   error
+	}{
+		"get VM credential": {
+			jwt:           thirdPartyJwt,
+			jwtPath:       fmt.Sprintf("/tmp/security-pkg-credentialfetcher-plugin-gcetest-%s", uuid.New().String()),
+			expectedToken: thirdPartyJwt,
+			expectedCall:  1,
+		},
+		"jwt path not set": {
+			jwt:          thirdPartyJwt,
+			expectedCall: 1,
+			expectedErr:  fmt.Errorf("jwtPath is unset"),
+		},
+		"fetch credential multiple times": {
+			expectedCall: 5,
+			jwtPath:      fmt.Sprintf("/tmp/security-pkg-credentialfetcher-plugin-gcetest-%s", uuid.New().String()),
+		},
+	}
+
+	SetTokenRotation(false)
+	ms, err := StartMetadataServer()
+	if err != nil {
+		t.Fatalf("StartMetadataServer() returns err: %v", err)
+	}
+	t.Cleanup(func() {
+		ms.Stop()
+		SetTokenRotation(true)
+	})
+
+	for id, tc := range testCases {
+		p := GCEPlugin{
+			tokenCache: tc.jwt,
+			jwtPath:    tc.jwtPath,
+		}
+		if err := creatJWTFile(tc.jwtPath); err != nil {
+			t.Fatalf("%s, creatJWTFile() returns err: %v", id, err)
+		}
+		ms.Reset()
+		ms.setToken(tc.jwt)
+
+		tokens, errs := getTokenFromServer(t, &p, tc.expectedCall)
+
+		verifyError(t, id, errs, tc.expectedErr)
+		if tc.expectedErr == nil {
+			if ms.NumGetTokenCall() != tc.expectedCall {
+				t.Errorf("%s, metadata server receives %d calls, want %d",
+					id, ms.NumGetTokenCall(), tc.expectedCall)
+			}
+			if tc.jwtPath != "" {
+				verifyToken(t, id, tc.jwtPath, tokens, tc.expectedToken)
+			}
+		}
+	}
+}
+
+func TestTokenRotationJob(t *testing.T) {
+	testCases := map[string]struct {
+		jwt           string
+		jwtPath       string
+		expectedToken string
+		expectedCall  int
+	}{
+		// mock metadata server returns an expired token, that forces rotation job
+		// to fetch new token during each rotation.
+		"expired token needs rotation": {
+			jwt:           thirdPartyJwt,
+			jwtPath:       fmt.Sprintf("/tmp/security-pkg-credentialfetcher-plugin-gcetest-%s", uuid.New().String()),
+			expectedToken: thirdPartyJwt,
+			expectedCall:  5,
+		},
+		// mock metadata server returns a token which has no exp field, that forces rotation job
+		// to fetch new token during each rotation.
+		"token with no expiration time needs rotation": {
+			jwt:           firstPartyJwt,
+			jwtPath:       fmt.Sprintf("/tmp/security-pkg-credentialfetcher-plugin-gcetest-%s", uuid.New().String()),
+			expectedToken: firstPartyJwt,
+			expectedCall:  5,
+		},
+		// mock metadata server returns an invalid token, that forces rotation job
+		// to fetch new token during each rotation.
+		"invalid token needs rotation": {
+			jwt:           "invalid-token-section-1.invalid-token-section-2",
+			jwtPath:       fmt.Sprintf("/tmp/security-pkg-credentialfetcher-plugin-gcetest-%s", uuid.New().String()),
+			expectedCall:  5,
+			expectedToken: "invalid-token-section-1.invalid-token-section-2",
+		},
+	}
+
+	// starts rotation job every 1 second.
+	rotationInterval = 1 * time.Second
```

Comment:
can you make this much smaller? the test takes 15s currently

Reply:
I have reduced the interval to 0.5 second, and reduced number of expected calls to 3, that's 4.5s now.